File: OOP C++/lab2/Sum.cpp
#include "Sum.h"
#include "FactoryInitializer.h"

void Sum::Execute(Operands& operands, Tokens& tokens, std::string& output, Reader& reader) {
    int operand1 = operands.GetAndPop();
    int operand2 = operands.GetAndPop();
    operands.Push(operand1 + operand2);
}

namespace {
    FactoryInitializer<Sum> Registration("+");
}

File: Lab1/src/Commands/Sum.java
package Commands;

import Utility.Operands;

import java.util.StringTokenizer;

public class Sum implements Command{
    @Override
    public void Execute(StringTokenizer tokens, Operands operands) {
        int num1 = operands.GetAndPop();
        int num2 = operands.GetAndPop();
        operands.Push(num1 + num2);
    }

    @Override
    public void Check(StringTokenizer tokens, Operands operands) {
    }

}
